### PR TITLE
Fix mount not being called sometimes

### PR DESCRIPTION
**Pull request by harwee**
## Description

Fix for issue https://github.com/imba/imba/issues/228

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by harwee** [comment]
I still don't this this is a perfect fix, `tryMount` and `tryUnmount` functions should be the one returning the count of mounted and unmounted tags that should be subtracted.

**Comment by harwee** [comment]
Count cannot keep proper track of tags that are mounted or should be unmounted since some tags are already mounted but passed to insert and while unmounting removal of one tag may result in removal of multiple mountable tags, so using boolean to track these changes seems to be much more stable than keeping count of number of tags that are inserted or removed. latest commit should be able to handle any case where any element that is inserted/removed while processing mounts/unmounts will be handled.

**Comment by somebee** [comment]
This is a good start which fixes imba/imba#228 but will break other cases. If you have a node tree with a mountable node nested inside it, and then later insert this outer node (which is not itself mountable) `_tryMounting` will not be set.

We really need some client-side tests for these behaviours.

**Comment by harwee** [comment]
Yes, I missed considering the scenario which you mentioned. Will update the PR with a fix soon.

**Comment by harwee** [comment]
I messed up with commit  09c6ad7 but it has been fixed in 152fcc8 you can review the code.

File: lib/imba/dom/manager.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ Imba.TagManagerClass.prototype.mounted = function (){
 };
 
 Imba.TagManagerClass.prototype.insert = function (node,parent){
-	this._inserts++;
+	if (node && node.querySelector && (node.mount || node.querySelector(".__mount"))) { this._tryMounting = true };
```
Comment (somebee) [inline lib/imba/dom/manager.js]: Essentially calling querySelector whenever a node is inserted somewhere in a tag tree in Imba will introduce a _hefty_ performance penalty, so we need to find a different (more optimized) approach. I will look at the test you added and try to think about some more performant ways of dealing with this :)

Comment (harwee) [reply lib/imba/dom/manager.js]: Yes, that is what I thought. But I couldn't think of any other way, so for now I will continue with this implementation until something better is found.

**Comment by somebee** [comment]
Quick update on this – there are quite a few challenges with efficient and correct mount/unmount behaviour. In the next version of Imba we have moved over to compiling tags to native web components, thereby getting efficient and correct mount/unmount in all cases :)

**Comment by somebee** [comment]
Closing this now since it is not relevant for the upcoming v2. Thank you so much for the effort anyway. Mount is finally consistently working in v2 🥳

**Pull request closed without merging**